mscore/scoreview.cpp

```diff
@@ -5494,6 +5495,7 @@ void ScoreView::cmdAddAnnotation2()
       annotation->setTrack(cr->track());
       annotation->textAnnotation()->setTextStyleType(TextStyleType::ANNOTATION);
       annotation->setParent(cr->segment());
+      cr->segment()->add(annotation);
```

Comment:
Look to see how other elements are added.  We seldom if ever call Segment::add() directly; it is usually Score::undoAddElement().  Look at ScoreView::cmdAddText() in particular for a model of how this is done.  Doing it this way is necessary to get things to work with linked parts and also with undo.  You will probably also need to add handling for your new element type within Score::undoAddElement().  Again, look to see how staff text is handled, and start by assuming a segment-attached annotation will need to do the same things.  The range-attached annotations will be considerably trickier, but we'll cross that bridge when we get there.